api/src/main/java/org/eclipse/microprofile/config/Config.java

```diff
@@ -181,6 +181,49 @@
         return getOptionalValue(propertyName, arrayType).map(Arrays::asList);
     }
 
+    /**
+     * Return the resolved configuration properties type with the specified prefix. 
+     * The type declaration can be annotated with 
+     * {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}. 
+     * If the type is annotated with {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}, 
+     * the prefix supplied in this method overrides the prefix associated with the annotation 
+     * {@linkplain org.eclipse.microprofile.config.inject.ConfigProperties @ConfigProperties}.
+     * 
+     * @param <T> 
+     *              The Class Type
+     * @param configProperties
+     *              The class that contains a number of fields that maps to corresponding configuration properties. 
```

Comment:
(I am trying to read this javadoc like a naïve user, on purpose.)  Does this mean that the class that is annotated with `@ConfigProperties` will have its _fields_ set to corresponding configuration property values?  Can the fields be of any access level (`public`, `protected`, `private`, etc.)?  Can they be of any type?  Is conversion involved (I presume so)?  Are covariant assignability semantics involved (I assume so)? (And so on.)

Reply:
> (I am trying to read this javadoc like a naïve user, on purpose.) Does this mean that the class that is annotated with `@ConfigProperties` will have its _fields_ set to corresponding configuration property values? 

Yes.

> Can the fields be of any access level (`public`, `protected`, `private`, etc.)? 

Yes.

> Can they be of any type? Is conversion involved (I presume so)? 

Yes.

> Are covariant assignability semantics involved (I assume so)? (And so on.)

We never discussed this. We may need to clarify it.

Reply:
This is to do with Converter. I suggest to raise a separate issue to clarify this becuase it affects all of the config property look not just here. If you have a convert to convert a string to Vehicle, can you use this to convert to a Car, which is a vehicle?